### ISSUE TITLE
error.errorCode: login_required not caught by TokenManger on error subscription

### ISSUE THREAD

**Issue by Savahn** [okta/okta-oidc-js#860]
## I'm submitting this issue for the package(s):

- [ ] jwt-verifier
- [ ] okta-angular
- [ ] oidc-middleware
- [x] okta-react
- [ ] okta-react-native
- [ ] okta-vue

## I'm submitting a:

- [x] Bug report 
- [ ] Feature request
- [ ] Other (Describe below)

## Current behavior
I am currently testing edge-case scenarios within our application, specifically around stale sessions and expired access tokens when the user is either on the app screen, or opening it back up after the token expiration. I have set my token policy to be a 5 minute token for testing purposes. Five minutes after logging in and receiving the token and navigating back to the app screen I expect that the user's token will be refreshed as I have not set autoRenew to be false. Instead, the page is blank for a long time before returning an OauthSdk timeout error, which I am able to capture with the tokenManager on error subscription however given the amount of time that elapses before this error is returned it is not a viable solution to my business case.

![image](https://user-images.githubusercontent.com/33760158/89802202-cd560c80-daf6-11ea-9f86-0364175e8a4f.png)

Before that error returns I see this error in the console, and I have as yet been unable to capture this error to either attempt to get a token from getTokenWithoutPrompt or send the user to the login screen (either would be acceptable in this scenario).

![image](https://user-images.githubusercontent.com/33760158/89802030-88ca7100-daf6-11ea-84ce-fdba65cfb452.png)

## Expected behavior
User token is either silently refreshed, or the error is able to be captured to redirect to the login screen.

## Environment

- Package Version:
  - okta-react: 3.0.4
  - okta-signin-widget: 4.2.1

**Comment by shuowu** [comment]
@Savahn I suspect you were running into a scenario where your OKTA SSO session got expired, but one of your tokens is still valid, then the current `okta-react` SDK will think you are still login by default, then still keep you in login status.

We are going to change the default behavior to set `isAuthenticated` to true only when both tokens are available in the next breaking change. Meanwhile, you can add a workaround by customize the `isAuthenticated` callback by following: https://github.com/okta/okta-oidc-js/tree/master/packages/okta-react#security

**Comment by Savahn** [comment]
I forgot to mention, I'm already using a workaround for the isAuthenticated callback.

```
isAuthenticated: async () => {
    const idToken = await authService.getTokenManager().get('idToken');
    const accessToken = await authService.getTokenManager().get('accessToken');
    return !!(idToken && accessToken);
  }
```

**Comment by shuowu** [comment]
@Savahn Can you share simple reproduction steps for your app, then I can look into the issue.

Also, I assume it may cause by two authClient instances running auto-renew process concurrently (one from the okta-react SDK, the other one from the sign-in-widget), I would recommend you turn off `autoRenew` in sign-in-widget by overriding the `authParams` configuration (https://github.com/okta/okta-signin-widget#configuration), then only maintain authState in `okta-react` SDK.

**Comment by Savahn** [comment]
Adding the autoRenew: false to the tokenManager in the authParams still leaves me with the same login_required error and the auth timeout. I'll attempt to get a minimalistic repo set up that recreates the issue. For now this is the auth config we use when setting up the sign-in widget, everything else is handled with useOktaAuth hooks from the okta-react package after sign-in.

```javascript
const { pkce, issuer, clientId, redirectUri, scopes } = config; //imported from a utilities file
const buildConfig = extras => ({
  baseUrl: issuer.split('/oauth2')[0],
  clientId,
  redirectUri, // implicit/callback
  i18n: {},
  authParams: {
    responseMode: pkce ? 'query' : 'fragment',
    issuer,
    display: 'page',
    scopes,
    state,
    pkce, // true
    tokenManager: {
      autoRenew: false
    }
  },
  idps: [], // We do have Google and Microsoft IDPs in place
  idpDisplay: 'PRIMARY',
  features: {}, // Registration is enabled as well, but the registration is a proprietary url and has not played a part in this issue
  registration: {},
  helpLinks: {},
  ...extras
});

/**
 * Okta Login page for AgConnections
 * @param {Object} props - component props
 */
const Authentication = ({ oktaConfig }) => {
  const signIn = new OktaSignIn(buildConfig(oktaConfig));
  useEffect(() => {
    try {
      signIn.renderEl({ el: '#sign-in-widget' });
    } catch (err) {
      signIn.show();
    }
  });

  return <div id="sign-in-widget" />;
};
Authentication.propTypes = {
  oktaConfig: PropTypes.objectOf(PropTypes.any)
};
Authentication.defaultProps = {
  oktaConfig: null
};

export default Authentication;
```

**Comment by shuowu** [comment]
Also, are you using the `SecureRoute` component from `okta-react` SDK in your app?

**Comment by Savahn** [comment]
Yes, we have a combination of `Route` and `SecureRoute` components

**Comment by shuowu** [comment]
@Savahn Your configuration looks good, and I also tried your SDK dependencies in the [okta react custom-login sample](https://github.com/okta/samples-js-react/tree/master/custom-login), looks like everything works well there.

As we discussed, I would suggest a minimalistic repo with reproduction steps can be helpful to identify your issue. Meanwhile, you can also reach out to our support team (developers@okta.com) to see if they can provide more detailed information for your issue.

**Comment by aarongranick-okta** [comment]
@Savahn The page loaded in the iframe (`/authorize`) will load HTML which should communicate back to the SDK running in your app. The timeout error indicates the HTML may not have loaded correctly. Is it possible this page is being passed through a proxy which may be modifying its contents or headers?

Load the page with the network inspector running and view the contents of the `/authorize` endpoint (made by the token renew call). It should be an HTML document with script (and not a 404 page). Normally, this page will emit its own errors to the console.
